lib/ocpp/v201/utils.cpp

```diff
@@ -157,7 +157,22 @@ std::string sha256(const std::string& str) {
 std::string generate_token_hash(const IdToken& token) {
     return sha256(conversions::id_token_enum_to_string(token.type) + token.idToken.get());
 }
+ocpp::DateTime round_to_x_seconds(const DateTime& timestamp, std::chrono::seconds x) {
+    auto timestamp_sys = date::utc_clock::to_sys(timestamp.to_time_point());
 
+    // get the current midnight
+    auto midnight = std::chrono::floor<date::days>(timestamp_sys);
+    auto seconds_since_midnight = std::chrono::duration_cast<std::chrono::seconds>(timestamp_sys - midnight).count();
```

Comment:
My suggestion is to remove all of the count kind of functions and stay within the chrono types